tests/pos/selectable-fields.scala

```diff
@@ -0,0 +1,3 @@
+val foo1 = new Selectable:
+  type Fields = (xyz: Int)
+  def selectDynamic(name: String): Any = 23
```

Comment:
you also need to test `foo1.xyz` because that's where the crash came from, I still see it in the repl:
```scala
scala> foo1.xyz

  unhandled exception while running erasure on rs$line$3

  An unhandled exception was thrown in the compiler.
  Please file a crash report here:
  https://github.com/scala/scala3/issues/new/choose
  For non-enriched exceptions, compile with -Yno-enrich-error-messages.

     while compiling: rs$line$3
        during phase: erasure
                mode: Mode(ImplicitsEnabled,ReadPositions,Interactive)
     library version: version 2.13.12
    compiler version: version 3.5.0-RC1-bin-SNAPSHOT-nonbootstrapped-git-164005a
            settings: -Ycook-docs true -Yread-docs true -classpath /Users/jamie/Workspace/dotty/out/bootstrap/scala3-compiler-bootstrapped/scala-3.5.0-RC1-bin-SNAPSHOT-nonbootstrapped/classes:/Users/jamie/Workspace/dotty/interfaces/target/classes:/Users/jamie/Workspace/dotty/out/bootstrap/scala3-library-bootstrapped/scala-3.5.0-RC1-bin-SNAPSHOT-nonbootstrapped/classes:/Users/jamie/Workspace/dotty/out/bootstrap/tasty-core-bootstrapped/scala-3.5.0-RC1-bin-SNAPSHOT-nonbootstrapped/classes:/Users/jamie/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/modules/scala-asm/9.6.0-scala-1/scala-asm-9.6.0-scala-1.jar:/Users/jamie/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-sbt/compiler-interface/1.9.6/compiler-interface-1.9.6.jar:/Users/jamie/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-reader/3.25.1/jline-reader-3.25.1.jar:/Users/jamie/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-terminal/3.25.1/jline-terminal-3.25.1.jar:/Users/jamie/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-terminal-jna/3.25.1/jline-terminal-jna-3.25.1.jar:/Users/jamie/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.12/scala-library-2.13.12.jar:/Users/jamie/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-sbt/util-interface/1.9.8/util-interface-1.9.8.jar:/Users/jamie/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-native/3.25.1/jline-native-3.25.1.jar:/Users/jamie/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/net/java/dev/jna/jna/5.14.0/jna-5.14.0.jar -d <REPL compilation output>

[error] java.lang.AssertionError: assertion failed: no owner from  <none>/ <none> in foo1.selectDynamic
[error] 	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
[error] 	at dotty.tools.dotc.transform.Erasure$Typer.typedSelect(Erasure.scala:719)
[error] 	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3177)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3285)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3363)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3367)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3478)
[error] 	at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:844)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3202)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3286)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3363)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3367)
[error] 	at dotty.tools.dotc.transform.Erasure$Typer.typedTyped(Erasure.scala:634)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3207)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3286)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3363)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3367)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3478)
[error] 	at dotty.tools.dotc.transform.Erasure$Typer.typedTypeApply(Erasure.scala:824)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3222)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3286)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3363)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3367)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3478)
[error] 	at dotty.tools.dotc.transform.Erasure$Typer.$anonfun$7(Erasure.scala:857)
[error] 	at dotty.tools.dotc.core.Decorators$.zipWithConserve(Decorators.scala:160)
[error] 	at dotty.tools.dotc.transform.Erasure$Typer.typedApply(Erasure.scala:857)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3202)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3286)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3363)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3367)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3478)
[error] 	at dotty.tools.dotc.transform.Erasure$Typer.typedTypeApply(Erasure.scala:824)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3222)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3286)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3363)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3367)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3478)
[error] 	at dotty.tools.dotc.typer.Typer.$anonfun$64(Typer.scala:2697)
[error] 	at dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:256)
[error] 	at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2697)
[error] 	at dotty.tools.dotc.transform.Erasure$Typer.typedDefDef(Erasure.scala:961)
[error] 	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3184)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3285)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3363)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3367)
[error] 	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3389)
[error] 	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3435)
[error] 	at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1074)
[error] 	at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:2884)
[error] 	at dotty.tools.dotc.transform.Erasure$Typer.typedClassDef(Erasure.scala:1050)
[error] 	at dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:3190)
[error] 	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3194)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3285)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3363)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3367)
[error] 	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3389)
[error] 	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3435)
[error] 	at dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1074)
[error] 	at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:3017)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3236)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3286)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3363)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3367)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3478)
[error] 	at dotty.tools.dotc.transform.Erasure.run(Erasure.scala:145)
[error] 	at dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:380)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] 	at scala.collection.immutable.List.foreach(List.scala:333)
[error] 	at dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:386)
[error] 	at dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:343)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] 	at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
[error] 	at dotty.tools.dotc.Run.runPhases$1(Run.scala:369)
[error] 	at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:384)
[error] 	at dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:396)
[error] 	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:69)
[error] 	at dotty.tools.dotc.Run.compileUnits(Run.scala:396)
[error] 	at dotty.tools.dotc.Run.compileUnits(Run.scala:288)
[error] 	at dotty.tools.repl.ReplCompiler.compile(ReplCompiler.scala:88)
[error] 	at dotty.tools.repl.ReplDriver.compile(ReplDriver.scala:321)
[error] 	at dotty.tools.repl.ReplDriver.interpret(ReplDriver.scala:283)
[error] 	at dotty.tools.repl.ReplDriver.loop$1(ReplDriver.scala:196)
[error] 	at dotty.tools.repl.ReplDriver.runUntilQuit$$anonfun$1(ReplDriver.scala:199)
[error] 	at dotty.tools.repl.ReplDriver.withRedirectedOutput(ReplDriver.scala:238)
[error] 	at dotty.tools.repl.ReplDriver.runBody$$anonfun$1(ReplDriver.scala:212)
[error] 	at dotty.tools.runner.ScalaClassLoader$.asContext(ScalaClassLoader.scala:80)
[error] 	at dotty.tools.repl.ReplDriver.runBody(ReplDriver.scala:212)
[error] 	at dotty.tools.repl.ReplDriver.runUntilQuit(ReplDriver.scala:199)
[error] 	at xsbt.ConsoleInterface.run(ConsoleInterface.java:52)
[error] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[error] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[error] 	at sbt.internal.inc.AnalyzingCompiler.invoke(AnalyzingCompiler.scala:329)
[error] 	at sbt.internal.inc.AnalyzingCompiler.console(AnalyzingCompiler.scala:233)
[error] 	at sbt.Console.console0$1(Console.scala:65)
[error] 	at sbt.Console.$anonfun$apply$5(Console.scala:75)
[error] 	at sbt.Run$.executeSuccess(Run.scala:187)
[error] 	at sbt.Console.$anonfun$apply$4(Console.scala:75)
[error] 	at sbt.internal.util.Terminal.withRawInput(Terminal.scala:146)
[error] 	at sbt.internal.util.Terminal.withRawInput$(Terminal.scala:144)
[error] 	at sbt.internal.util.Terminal$ProxyTerminal$.withRawInput(Terminal.scala:424)
[error] 	at sbt.Console.$anonfun$apply$3(Console.scala:75)
[error] 	at sbt.internal.util.Terminal$TerminalImpl.withRawOutput(Terminal.scala:1028)
[error] 	at sbt.internal.util.Terminal$ProxyTerminal$.withRawOutput(Terminal.scala:463)
[error] 	at sbt.Console.apply(Console.scala:72)
[error] 	at sbt.Console.apply(Console.scala:50)
[error] 	at sbt.Console.apply(Console.scala:42)
[error] 	at sbt.Defaults$.$anonfun$consoleTask$1(Defaults.scala:2237)
[error] 	at sbt.Defaults$.$anonfun$consoleTask$1$adapted(Defaults.scala:2223)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error] 	at sbt.Execute.work(Execute.scala:292)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
[error] 	at java.base/java.lang.Thread.run(Thread.java:1583)
[error] (scala3-compiler-bootstrapped / Compile / console) java.lang.AssertionError: assertion failed: no owner from  <none>/ <none> in foo1.selectDynamic
[error] Total time: 155 s (02:35), completed 26 Apr 2024, 23:04:16
```

Reply:
OK, got it now. It's now a static error.

Reply:
nice!